bundler/lib/bundler/settings.rb

```diff
@@ -595,5 +596,31 @@ def self.key_to_s(key)
         end
       end
     end
+
+    def credentials_from_helper(uri)
+      helper_key = "credential.helper.#{uri.host}"
+      helper_path = self[helper_key]
+      return unless helper_path
+
+      begin
+        require "shellwords"
+        command = Shellwords.shellsplit(helper_path)
+        command[0] = if command[0].start_with?("/", "~")
+          command[0]
+        else
+          "bundler-credential-#{command[0]}"
+        end
+
+        output = Bundler.with_unbundled_env { IO.popen(command, &:read) }
```

Comment:
nit - lets check `$?` after `IO.popen`

Reply:
thanks, db1e5abb1c fixed checking Process.last_status (use $? difficult to write specs)